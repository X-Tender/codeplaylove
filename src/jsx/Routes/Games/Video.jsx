import React from 'react';
import PropTypes from 'prop-types';
import ResponsiveEmbed from 'react-responsive-embed';

const propTypes = {
	youtubeId: PropTypes.string.isRequired,
	className: PropTypes.string,
	ratio: PropTypes.string,
};

const defaultProps = {
	className: null,
	ratio: '4:3',
};

const Video = ({ youtubeId, className, ratio }) => (
	<div className={className} style={{ gridColumn: '1/-1' }}>
		<ResponsiveEmbed
			allowFullScreen
			ratio={ratio}
			src={`https://www.youtube-nocookie.com/embed/${youtubeId}?rel=0&amp;showinfo=0&theme=white`}
		/>
	</div>
);

Video.propTypes = propTypes;
Video.defaultProps = defaultProps;

export default Video;
