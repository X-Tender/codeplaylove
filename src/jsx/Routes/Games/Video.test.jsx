import React from 'react';
import { shallow } from 'enzyme';
import Video from './Video';

describe('Video', () => {
	const youtubeId = 'fJ9rUzIMcZQ';
	const className = 'we-will-rock-you';

	const comp = shallow(<Video className={className} youtubeId={youtubeId} />);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('renders props correctly', () => {
		expect(comp.find(`.${className}`).exists()).toBe(true);
	});
});
