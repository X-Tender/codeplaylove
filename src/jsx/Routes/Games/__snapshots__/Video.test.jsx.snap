// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders correctly 1`] = `
<div
  className="we-will-rock-you"
  style={
    Object {
      "gridColumn": "1/-1",
    }
  }
>
  <ResponsiveEmbed
    allowFullScreen={true}
    ratio="4:3"
    src="https://www.youtube-nocookie.com/embed/fJ9rUzIMcZQ?rel=0&amp;showinfo=0&theme=white"
  />
</div>
`;
